pkg/logicalrestore/logicalrestore.go

```diff
@@ -83,14 +40,17 @@ type LogicalRestore struct {
 	ctx  context.Context
 
 	baseDir string
+	tblOid  dbutils.Oid
+	curTx   int32
+	curLsn  dbutils.Lsn
 }
 
-func New(schemaName, tableName, dir string, cfg pgx.ConnConfig) *LogicalRestore {
+func New(tbl message.NamespacedName, dir string, cfg pgx.ConnConfig) *LogicalRestore {
```

Comment:
Looks like the restorer aims to do both the context management and general setup and table restore. Perhaps it makes sense to split it into the manager class, that handles shutdown and spawns worker and the restorer worker tasked to restore a single table (would work better for the case of restoring multiple tables).